cmd/createcluster.go

```
@@ -386,43 +398,94 @@ func getValidators(dvs []tbls.TSS) ([]cluster.DistValidator, error) {
 	return vals, nil
 }
 
-// getOperators returns a list of operators from the provided secret keyshares.
-func getOperators(numNodes int, clusterDir string, insecureKeys bool, shareSets [][]*bls_sig.SecretKeyShare) ([]cluster.Operator, error) {
-	var peers []p2p.Peer
+// saveKeysToKeyManager saves validator keys to the provided keymanager addresses.
+func saveKeysToKeymanager(ctx context.Context, addrs []string, numNodes int, shareSets [][]*bls_sig.SecretKeyShare) error {
+	if len(addrs) != numNodes {
+		return errors.New("insufficient no of keymanager addresses", z.Int("expected", numNodes), z.Int("got", len(addrs)))
```

fail fast, do config validation at the start of the process.